In [ ]:
import agate

this code for importing the agate. first step. 

In [2]:
salaries = agate.Table.from_csv('Data/nusalaries1718.csv')

Next step: Importing data. We use the code ('Data/nusalaries1718.csv') because the data is saved in the data folder, home directory. If you put in the download use code ('/Users/utamidiahkusumawati/Downloads/nusalaries1718.csv')

In [3]:
print(salaries)

| column                        | data_type |
| ----------------------------- | --------- |
| Employee                      | Text      |
| Position                      | Text      |
| Campus                        | Text      |
| Department                    | Text      |
| Budgeted Annual Salary        | Number    |
| Salary from State Aided Funds | Number    |
| Salary from Other Funds       | Number    |



So, let's print the data by using code print(variable name) --> print(salaries)

In [4]:
salaries.print_table()

| Employee             | Position             | Campus | Department           | Budgeted Annual S... | Salary from State... | ... |
| -------------------- | -------------------- | ------ | -------------------- | -------------------- | -------------------- | --- |
| Abbey, Bryce M       | Assistant Professor  | UNK    | Kinesiology & Spo... |               52,625 |               52,625 | ... |
| Abbott, Amanda A     | Admissions Associate | UNO    | Graduate Studies     |               34,164 |               34,164 | ... |
| Abbott, Frances M    | Staff Secy III       | UNL    | FM&P Facilities P... |               36,676 |               36,676 | ... |
| Abboud, Cheryl A     | Administrator I      | UNMC   | Surgery-Urologic ... |               72,774 |                      | ... |
| Abdalla, Maher Y     | Asst Professor       | UNMC   | Pathology/Microbi... |               73,488 |               58,866 | ... |
| Abdel-Monem, Tarik L | Research Specialist  | UNL    | Public Policy Cente

If you use the code: print(salaries) or salaries.print_structure(), you will not get the whole pictures of the tables, including the columns and rows, numbers and texts. However, if you want to know whole pictures of table, with numbers or long texts, type code: variable name.print_table() --> salaries.print_table()

In [6]:
salaries.print_structure()

| column                        | data_type |
| ----------------------------- | --------- |
| Employee                      | Text      |
| Position                      | Text      |
| Campus                        | Text      |
| Department                    | Text      |
| Budgeted Annual Salary        | Number    |
| Salary from State Aided Funds | Number    |
| Salary from Other Funds       | Number    |


test case for knowing what kind of columns, same with previous step.

In [8]:
unl = salaries.where(lambda row: row['Campus'] is 'UNL')

Since the table provides you with salaries from all University of Nebraska campus, you need to limit it only for UNL employee. So.... let's call the variable by using code: unl = salaries.where(lambda row: row('Campus') is 'UNL') Ok, campus and UNL are the name of values we choose, and you use table.where() to pick up certain rows. (actually im still confused why we dont use:   table.select(lambda column: column('Campus')...? since we are taking columns rather than rows??? #question 

In [9]:
print (len(unl.rows))

0


this is the code for filtering UNL employee only salaries. (PS. I still do not know what len refers to and why should rows and not columns?)

In [10]:
unl = salaries.where(lambda row: row['Campus'] == 'UNL')

it seems that you should add one more equal sign == because UNL is not variable.

In [2]:
old = len (salaries.rows)

NameError: name 'salaries' is not defined

In [11]:
print (len(unl.rows))

6315


call the result out

In [12]:
average_salaries = salaries.aggregate(agate.Mean('Budgeted Annual Salary'))

finding average (mean) by using this template of code: name = variable.aggregate(agate.Mean('value')). ----> average_salaries = salaries.aggregate(agate.Mean('Budgeted Annual Salary')). 

In [13]:
print(average_salaries)

60972.20300236226472605349387


call the result out!

In [14]:
median_salaries = salaries.aggregate(agate.Median('Budgeted Annual Salary'))

change the code with median. the rest is the same, hope this is correct one to count. 

In [15]:
print(median_salaries)

50247


the result of median annual salaries of the UNL employees. $50,247

In [3]:
by_position = unl.group_by('Position')

NameError: name 'unl' is not defined